daemon/image_pull.go

```diff
@@ -104,3 +104,39 @@ func (daemon *Daemon) pullImageWithReference(ctx context.Context, ref reference.
 	<-writesDone
 	return err
 }
+
+func (daemon *Daemon) ResolveTagToDigest(ctx context.Context, ref reference.NamedTagged, authConfig *types.AuthConfig) (digest.Digest, error) {
```

Comment:
Rather than add yet another method to daemon, why not export access to the configured repository? As it is, this method is very narrow and not very flexible. Ideally, we give it an image reference and get back fully configured repository.